jenkins/opensearch-dashboards/Jenkinsfile

```diff
@@ -164,7 +174,9 @@ pipeline {
                     publishNotification(
                         icon: ':white_check_mark:',
                         message: 'Successful Build',
-                        extra: stashed
+                        extra: stashed,
+                        credentialsId: 'BUILD_NOTICE_WEBHOOK',
+                        variable: 'BUILD_NOTICE_WEBHOOK'
```

Comment:
No need to expose `variable`, `withCredentials` takes a name of this variable to assign the value to, and then we use it inside its block. Rename it to something more common like `WEBHOOK_URL` inside `publishNotification`.

Reply:
We have different webhooks for build as `BUILD_NOTICE_WEBHOOK` and for integ test as `INTEG_TEST_WEBHOOK`. The reason being having 2 different channel to send notification. Is there anything I'm missing here?

Reply:
Yes, you're missing the fact that the variable used is purely internal to inside of the `withCredential` block. The following produces the same output

```
withCredentials([string(credentialsId: 'THIS_IS_THE_ID_OF_CREDENTIALS', variable: 'FOO')]) {
  echo $FOO
}

withCredentials([string(credentialsId: 'THIS_IS_THE_ID_OF_CREDENTIALS', variable: 'BAR')]) {
  echo $BAR
}

```

Reply:
Also, I'm not sure but should we create separate global files for test instead of passing so many parameters because of different use cases? Let me know WDYT

Reply:
I could be wrong, but AFAIK the ID is different, but the name of the variable that receives the value can be anything.

![Screen Shot 2021-11-19 at 1 06 15 PM](https://user-images.githubusercontent.com/542335/142671429-185c020f-c097-466a-b83a-b2cd16f2efc0.png)

So 

```
withCredentials([string(credentialsId: 'ID', variable: 'FOO')]) {
   echo $FOO
}
```

produces the same output as 

```
withCredentials([string(credentialsId: 'ID', variable: 'BAR')]) {
   echo $BAR
}
```

Reply:
This worked for test notification. Trying it for build notification to see if the notification is triggered. Thanks for the input.

Reply:
Update: Worked for build as well.